components/constellation/constellation.rs

```diff
@@ -715,13 +717,24 @@ where
 
     /// The main event loop for the constellation.
     fn run(&mut self) {
+        // Start a fetch thread.
+        // In single-process mode this will be the global fetch thread;
+        // in multi-process mode this will be used only by the canvas paint thread.
+        let join_handle = start_fetch_thread(&self.public_resource_threads.core_thread);
```

Comment:
The use of `public_resource_threads` is quite interesting because it means that in single process mode, we'll always use the public resource threads for all fetches. 

The current code instead uses a kind of race condition, depending on who is first to initialize the fetch thread. For example, in single process mode, if the canvas paint thread is first to do an async fetch, then the global fetch thread will use the public resource thread. If on the other hand a pipeline is first, then the global fetch thread will use the result of [the resource thread selected as part of pipeline creation](https://github.com/servo/servo/blob/5e89f79abe5020e535934e1c896c0b6bdb5a3929/components/constellation/constellation.rs#L912)

Reply:
This can be fixed by making by removing the core resource thread state from the fetch thread, as done in https://github.com/servo/servo/pull/38422/commits/b7eabb635c55e2d1be7979ab09e808824bf4758e